File: 0019-remove-nth-node-from-end-of-list/0019-remove-nth-node-from-end-of-list.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def removeNthFromEnd(self, head: Optional[ListNode], n: int) -> Optional[ListNode]:
        count = 0 
        ptr = head 
        while ptr:
            ptr = ptr.next
            count+=1
        if count == n :
            return head.next
        count = count - n 
        ptr = head
        while count != 1 :
            count-=1
            ptr = ptr.next
        ptr.next = ptr.next.next
        return head 

File: 0019-remove-nth-node-from-end-of-list/0019-remove-nth-node-from-end-of-list.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode() : val(0), next(nullptr) {}
 *     ListNode(int x) : val(x), next(nullptr) {}
 *     ListNode(int x, ListNode *next) : val(x), next(next) {}
 * };
 */
class Solution {
public:
    ListNode* removeNthFromEnd(ListNode* head, int n) {
        if(head->next == nullptr) return nullptr;
        ListNode * ptr1=head, *ptr2=head;
        for(int i = 0; i<n; i++){
            ptr2 = ptr2->next;
        }
        if(ptr2 == nullptr) return head->next;
        while(ptr2->next != nullptr){
            ptr1 = ptr1->next;
            ptr2 = ptr2->next;
        }
        ptr1->next = ptr1->next->next;
        return head;
        
    }
};